### PR TITLE
e2e: Use the intersection of images available for windows and ubuntu to determine latest patch

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: While new patch releases had been published in the capi offer for Ubuntu and the Windows offer publishing was still ongoing, some tests started failing with `The platform image 'cncf-upstream:capi-windows:k8s-1dot22dot6-windows-2019-containerd:latest is not available`. This is because the tests were selecting the latest available patch release for Ubuntu, but there wasn't an equivalent windows image yet.

This PR tweaks the logic in the e2e test helper to make sure we select the latest version available for _both_ windows and linux.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/hold

let's merge #1996 first to unblock PRs

**Comment by CecileRobertMichon** [comment]
/retest

I was able to track down the previous VM create failures which were due to 

```
{"statusMessage":"{\"status\":\"Failed\",\"error\":{\"code\":\"ResourceOperationFailure\",\"message\":\"The resource operation completed with terminal provisioning state 'Failed'.\",\"details\":[{\"code\":\"OSProvisioningTimedOut\",\"message\":\"OS Provisioning for VM 'capz-e2e-fntc2' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. For details on how to check current provisioning state of Windows VMs, refer to https://aka.ms/WindowsVMLifecycle and Linux VMs, refer to https://aka.ms/LinuxVMLifecycle.\"}]}}"
```

this is a different error than the previous image not found error but it may indicate an issue with the image being listed but not fully replicated yet. Let's see what the retest says.

**Comment by jsturtevant** [comment]
I looked into this a bit and it appears that the latest Windows1.22.6 VHD image is invalid.  Testing with an updated image locally now

**Comment by jsturtevant** [comment]
confirmed this is caused by the Windows image being built with the windows patch [January 11, 2022—KB5009557 (OS Build 17763.2452)](https://support.microsoft.com/en-us/topic/january-11-2022-kb5009557-os-build-17763-2452-c3ee4073-1e7f-488b-86c9-d050672437ae) which has known issues.  There was an [Out-of-band patch January 18, 2022—KB5010791 OS Build 17763.2458 Out-of-band](https://support.microsoft.com/en-us/topic/january-18-2022-kb5010791-os-build-17763-2458-out-of-band-43697313-d8e0-4918-b6df-7f64d4d9a8cd) and we are in process of republishing the images with the new windows patch

**Comment by jsturtevant** [comment]
the new images are published
/retest

**Review by jackfrancis — APPROVED** [review]
lgtm

File: test/e2e/e2e_suite_test.go
Comment (jsturtevant) [inline test/e2e/e2e_suite_test.go:449]: the order of these args is different the `imagesClient.ListSkus(ctx, location, publisher, offer)` which was a bit confusing when reading. This is minor but could help clarity.

File: test/e2e/config/azure-dev.yaml
Comment (jsturtevant) [inline test/e2e/config/azure-dev.yaml:179]: Will this select the same version we are already on since both deployment and upgrade would be the same (`stable-1.22`)?

**Review by jsturtevant — COMMENTED** [review]
generally, looks good. couple minor questions/comments

File: test/e2e/e2e_suite_test.go
Comment (CecileRobertMichon) [reply test/e2e/e2e_suite_test.go:449]: true, let me fix that to make them match

File: test/e2e/config/azure-dev.yaml
Comment (CecileRobertMichon) [reply test/e2e/config/azure-dev.yaml:179]: upgrade from is set to 1.22.1 so it would upgrade from 1.22.1 to 1.22.6 (or latest)

I believe we had to set ugrade from to 1.22 (used to be 1.21) in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1792, we might be able to open a separate PR to revert it now?

Comment (jsturtevant) [reply test/e2e/config/azure-dev.yaml:179]: ah, missed that was a separate configuration. Thought it was from `KUBERNETES_VERSION`.

We can't revert as we still need 1.22 for windows but we could set `KUBERNETES_VERSION_UPGRADE_TO` to 1.23 now that it is released.

Comment (CecileRobertMichon) [reply test/e2e/config/azure-dev.yaml:179]: I'll open a separate PR to change upgrade, need to change the worker count as well since CAPI 1.1 beta merged

File: test/e2e/e2e_suite_test.go
Comment (CecileRobertMichon) [reply test/e2e/e2e_suite_test.go]: fixed

**Comment by CecileRobertMichon** [comment]
/retest

and 

/hold - this was another OS provisioning timeout failure for the two windows VMs

**Comment by CecileRobertMichon** [comment]
the windows VMs created successfully this time. The private cluster spec failed because the Azure Bastion took a long time to create.

/retest

**Review by mboersma — APPROVED** [review]
/lgtm

**Comment by CecileRobertMichon** [comment]
run into the windows OS provisioning timeout flake again.

Given how flaky this is, let's hold this until we have a new image with the December patch so we don't run into many flakes in PRs @mboersma @jsturtevant 

/hold

**Comment by CecileRobertMichon** [comment]
new image is live

/retest

**Pull request merged**
